src/backend/distributed/executor/executor.c

```diff
@@ -1565,6 +1575,26 @@ ManageWorkerPool(WorkerPool *workerPool)
 		 */
 		newConnectionCount = Min(readyTaskCount - idleConnectionCount, targetPoolSize) -
 							 activeConnectionCount;
+
+		if (newConnectionCount > 0 && ExecutorSlowStartInterval > 0)
+		{
+			TimestampTz now = GetCurrentTimestamp();
+
+			if (TimestampDifferenceExceeds(workerPool->lastConnectionOpenTime, now,
+										   ExecutorSlowStartInterval))
+			{
+				newConnectionCount = Min(newConnectionCount,
+										 workerPool->maxNewConnectionsPerCycle);
+
+				/* exponentially increase the open rate (like TCP slow start) */
+				workerPool->maxNewConnectionsPerCycle *= 2;
```

Comment:
wondering if we should start doubling from the second attempt.

e.g. if we do 1+1+2+4+8+16+32, can start utilising 32 cores after 64 cores after 60ms. If we do 1+2+4+8+16+32 we get this slightly odd result of having 63 connections after 50ms and 64 after 60ms.